src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -327,8 +325,12 @@ internal bool IsRuntimeAsyncEnabledIn(MethodSymbol method)
                 return false;
             }
 
-            // PROTOTYPE: Check for attributes that turn on/off the feature member-by-member
-            return true;
+            return method switch
+            {
+                SourceMethodSymbol { IsRuntimeAsyncEnabledInMethod: ThreeState.True } => true,
+                SourceMethodSymbol { IsRuntimeAsyncEnabledInMethod: ThreeState.False } => false,
+                _ => Feature("runtime-async") == "on"
```

Comment:
It looks like the default (on a new runtime) is still to generate the old state machine. Is that what we want?
I think the default should be to use runtime-async when the runtime supports it and there is no other gesture. That behavior is closer to the behavior we'll want after preview period.
Then you can make other gestures to control that more: 1. you can opt-out at method level with attribute, or 2. you can opt-out globally then restore only the methods that you want.

Reply:
This isn't what I remember agreeing on though: from what I recall, the feature flag will be there in .NET 10. We do not want to turn this on by default in .NET 10, so explicit opt-in is needed.

Reply:
Ah, you're right. Given that we're not using a forked runtime and that we'll merge the preview compiler feature, we need an explicit opt-in... Thanks